File: DataSummitSolution/DataSummitWeb/DTO/Rectangle.cs


namespace DataSummitWeb.DTO
{
    public class Rectangle
    {
        public Point Tl { get; set; }
        public Point Tr { get; set; }
        public Point Bl { get; set; }
        public Point Br { get; set; }
        public long Width { get; set; }
        public long Height { get; set; }
        public string Fill { get; set; }
        public string Type { get; set; }

        public Rectangle()
        { }
    }
}

File: DataSummitSolution/DataSummitUI/src/app/projects/models/rectangle.ts
import { Point } from "./point"

export class Rectangle {

    Tl: Point
    Tr: Point
    Bl: Point
    Br: Point
    Width: number
    Height: number
    Fill: string
    Type: string 

    constructor(tl?: Point, tr?: Point, bl?: Point, br?: Point,
                width?: number, height?: number, fill?: string, type?: string) {
        this.Tl = tl || new Point()
        this.Tr = tr || new Point()
        this.Bl = bl || new Point()
        this.Br = br || new Point()
        this.Width = width || 0
        this.Height = height || 0
        this.Fill = fill || ""
        this.Type = type || ""
    }
}